src/researches/findKeywordInFirstParagraph.js

```diff
@@ -16,13 +19,44 @@ import { isEmpty } from "lodash-es";
  * @returns {boolean} True if the text consists only of images, false otherwise.
  */
 function paragraphConsistsOfImagesOnly( text ) {
+	// Remove links from the text
+	const anchors = getAnchorsFromText( text );
+	if ( anchors.length > 0 ) {
+		anchors.forEach( function( anchor ) {
+			text = text.replace( anchor, "" );
+		} );
+
+		if ( text === "" ) {
+			return true;
+		}
+	}
+
```

Comment:
The block of code above may be a candidate for a little private function (in this file) 😉 .
Just to remove the anchor tags from the text. Then afterwards check if the text is empty after the function was applied.